packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -31,6 +33,134 @@ typedef void _RegisterServiceExtensionCallback({
+class _ProxyLayer<L extends Layer> extends Layer {
+  final L _layer;
+
+  _ProxyLayer(this._layer);
+
+  @override
+  void addToScene(ui.SceneBuilder builder, Offset layerOffset) {
+    _layer.addToScene(builder, layerOffset);
+  }
+
+  @override
+  S find<S>(Offset regionOffset) => _layer.find(regionOffset);
+}
+
+/// A layer that renders like an [OffsetLayer] with the offset modified.
+///
+/// The [OffsetLayer] interface is not implemented as generally all methods
+/// would need to throw an exception to avoid accidentally modifying the layer
+/// being proxied.
+class _OffsetProxyLayer extends _ProxyLayer<OffsetLayer> {
+  final Offset _offset;
+
+  _OffsetProxyLayer(OffsetLayer _layer, this._offset) : super(_layer);
+
+  Offset _transformOffset(Offset offset) => offset + _offset - _layer.offset;
+
+  @override
+  void addToScene(ui.SceneBuilder builder, Offset layerOffset) {
+    _layer.addToScene(builder, _transformOffset(layerOffset));
+  }
+
+  @override
+  S find<S>(Offset regionOffset) {
+    return _layer.find(_transformOffset(regionOffset));
+  }
+}
+
+/// A place to paint to create screenshots of [RenderObject]s.
+///
+/// Requires that the render objects have already been painted successfully as
+/// part of the regular rendering pipeline.
+class _ScreenshotPaintingContext extends PaintingContext {
+  _ScreenshotPaintingContext._(
+    ContainerLayer containerLayer,
+    Rect estimatedBounds,
+  ) : super(containerLayer, estimatedBounds);
+
+  @override
+  void compositeLayerAtOffset(OffsetLayer layer, Offset offset) {
+    // Wrap the layer in a proxy layer to avoid detaching the layer from its
+    // existing parent or modifying its offset.
+    appendLayer(new _OffsetProxyLayer(layer, offset));
+  }
+
+  @override
+  void addLayer(Layer layer) {
+    // Wrap the layer in a proxy layer to avoid detaching the layer from its
+    // existing parent.
+    super.addLayer(new _ProxyLayer<Layer>(layer));
+  }
+
+  @override
+  PaintingContext createChildContext(ContainerLayer childLayer, Rect bounds) {
+    return new _ScreenshotPaintingContext._(childLayer, bounds);
+  }
+
+  /// Captures an image of the current state of [renderObject] and its children.
+  ///
+  /// The returned [ui.Image] has uncompressed raw RGBA bytes, will be offset
+  /// by the top-left corner of [renderBounds], and have dimensions equal to the
+  /// size of [renderBounds] multiplied by [pixelRatio].
+  ///
+  /// To use [toImage], the render object must have gone through the paint phase
+  /// (i.e. [debugNeedsPaint] must be false).
+  ///
+  /// The [pixelRatio] describes the scale between the logical pixels and the
+  /// size of the output image. It is independent of the
+  /// [window.devicePixelRatio] for the device, so specifying 1.0 (the default)
+  /// will give you a 1:1 mapping between logical pixels and the output pixels
+  /// in the image.
+  ///
+  /// [debugPaint] specifies whether the image should include the debug paint
```

Comment:
to avoid sentences starting with a lower case letter, consider `The [debugPaint] argument specifies...`

Reply:
"the debug paint" is not clear to people who don't know what debugPaint means. Maybe like to [RenderObject.debugPaint] or something.

Reply:
done